jscomp/syntax/src/res_printer.ml

```diff
@@ -1841,6 +1845,8 @@ and print_typ_expr ~(state : State.t) (typ_expr : Parsetree.core_type) cmt_tbl =
   in
   let should_print_its_own_attributes =
     match typ_expr.ptyp_desc with
+    | Ptyp_constr _ when Ast_uncurried.core_type_is_uncurried_fun typ_expr ->
```

Comment:
Let the attribute printing handled by `print_typ_expr`, otherwise it's gonna cause duplicated attributes.

Reply:
We need to add a task to remove dead code once uncurried is the only possible function type